crates/quick_action_bar/src/quick_action_bar.rs

```diff
@@ -16,31 +17,55 @@ pub struct QuickActionBar {
     active_item: Option<Box<dyn ItemHandle>>,
     _inlay_hints_enabled_subscription: Option<Subscription>,
     workspace: WeakView<Workspace>,
+    show: bool,
 }
 
 impl QuickActionBar {
-    pub fn new(buffer_search_bar: View<BufferSearchBar>, workspace: &Workspace) -> Self {
-        Self {
+    pub fn new(
+        buffer_search_bar: View<BufferSearchBar>,
+        workspace: &Workspace,
+        cx: &mut ViewContext<Self>,
+    ) -> Self {
+        let mut this = Self {
             buffer_search_bar,
             active_item: None,
             _inlay_hints_enabled_subscription: None,
             workspace: workspace.weak_handle(),
-        }
+            show: true,
+        };
+        this.apply_settings(cx);
+        cx.observe_global::<SettingsStore>(|this, cx| this.apply_settings(cx))
```

Comment:
This will trigger on every settings access elsewhere and on every non-related settings change, so we might avoid sending redundant events here by ensuring that `self.show` and the new value are actually different.

Reply:
You are right. Thanks for noticing! Fixed.